src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RenameEndpoint.cs

```diff
@@ -88,13 +115,12 @@ internal class RenameEndpoint : IRenameEndpoint
             }
 
             var requestDocumentSnapshot = documentContext.Snapshot;
-
             if (!FileKinds.IsComponent(requestDocumentSnapshot.FileKind))
```

Comment:
This is one of the optimizations. It checks for a component. Is that normal?

Reply:
Ah interesting. It can be due to .razor and .cshtml being different tooling experiences. That being said I do wonder how much we care about optimizing this scenario for rename. Meaning, if a user went out of their way to rename something they hit `F2` (or equivalent) and doing a little bit extra work might be ok

Reply:
I didn't notice this when doing the rename, but this is actually probably a bug and should be removed. This check means that I inadvertently stopped people being able to trigger C# renames from .cshtml files

Reply:
Cool. I guess it's fixed then :)